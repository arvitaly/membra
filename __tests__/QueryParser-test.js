"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const schema2_1 = require("./../__fixtures__/schema2");
const QueryParser_1 = require("./../QueryParser");
// const schema = buildClientSchema(JSON.parse(readFileSync(__dirname + "/../__fixtures__/schema.json", "utf8")).data);
describe("QueryParser", () => {
    it("simple", () => {
        const parser = new QueryParser_1.default(schema2_1.default);
        const query = parser.parse `
            query Q1{
                viewer{
                    model1{
                        edges{
                            node{
                                field1
                            }
                        }
                    }
                }
            }
        `;
        expect(query).toMatchSnapshot();
    });
});
