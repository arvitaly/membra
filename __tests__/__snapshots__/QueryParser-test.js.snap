exports[`QueryParser simple 1`] = `
Object {
  "fields": Array [
    Object {
      "args": Array [],
      "fields": Array [
        Object {
          "args": Array [],
          "fields": Array [
            Object {
              "args": Array [],
              "fields": Array [
                Object {
                  "args": Array [],
                  "fields": Array [
                    Object {
                      "args": Array [],
                      "fields": Array [],
                      "isConnection": false,
                      "isNode": false,
                      "name": "field1",
                      "node": Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "loc": Object {
                          "end": 178,
                          "start": 172,
                        },
                        "name": Object {
                          "kind": "Name",
                          "loc": Object {
                            "end": 178,
                            "start": 172,
                          },
                          "value": "field1",
                        },
                        "selectionSet": null,
                      },
                      "type": "String",
                    },
                    Object {
                      "args": Array [],
                      "fields": Array [],
                      "isConnection": false,
                      "isFragment": false,
                      "isNode": false,
                      "name": "id",
                      "node": Object {
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                      },
                      "type": "ID!",
                    },
                  ],
                  "isConnection": false,
                  "isNode": true,
                  "name": "node",
                  "node": Object {
                    "alias": null,
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "loc": Object {
                      "end": 208,
                      "start": 134,
                    },
                    "name": Object {
                      "kind": "Name",
                      "loc": Object {
                        "end": 138,
                        "start": 134,
                      },
                      "value": "node",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "loc": Object {
                        "end": 208,
                        "start": 138,
                      },
                      "selections": Array [
                        Object {
                          "alias": null,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "loc": Object {
                            "end": 178,
                            "start": 172,
                          },
                          "name": Object {
                            "kind": "Name",
                            "loc": Object {
                              "end": 178,
                              "start": 172,
                            },
                            "value": "field1",
                          },
                          "selectionSet": null,
                        },
                        Object {
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "id",
                          },
                        },
                      ],
                    },
                  },
                  "type": "Model1",
                },
              ],
              "isConnection": false,
              "isNode": false,
              "name": "edges",
              "node": Object {
                "alias": null,
                "arguments": Array [],
                "directives": Array [],
                "kind": "Field",
                "loc": Object {
                  "end": 234,
                  "start": 99,
                },
                "name": Object {
                  "kind": "Name",
                  "loc": Object {
                    "end": 104,
                    "start": 99,
                  },
                  "value": "edges",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "loc": Object {
                    "end": 234,
                    "start": 104,
                  },
                  "selections": Array [
                    Object {
                      "alias": null,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "loc": Object {
                        "end": 208,
                        "start": 134,
                      },
                      "name": Object {
                        "kind": "Name",
                        "loc": Object {
                          "end": 138,
                          "start": 134,
                        },
                        "value": "node",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "loc": Object {
                          "end": 208,
                          "start": 138,
                        },
                        "selections": Array [
                          Object {
                            "alias": null,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "loc": Object {
                              "end": 178,
                              "start": 172,
                            },
                            "name": Object {
                              "kind": "Name",
                              "loc": Object {
                                "end": 178,
                                "start": 172,
                              },
                              "value": "field1",
                            },
                            "selectionSet": null,
                          },
                          Object {
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                          },
                        ],
                      },
                    },
                  ],
                },
              },
              "type": "[Model1ConnectionEdge]!",
            },
          ],
          "isConnection": true,
          "isNode": false,
          "name": "model1",
          "node": Object {
            "alias": null,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "loc": Object {
              "end": 256,
              "start": 67,
            },
            "name": Object {
              "kind": "Name",
              "loc": Object {
                "end": 73,
                "start": 67,
              },
              "value": "model1",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "loc": Object {
                "end": 256,
                "start": 73,
              },
              "selections": Array [
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "loc": Object {
                    "end": 234,
                    "start": 99,
                  },
                  "name": Object {
                    "kind": "Name",
                    "loc": Object {
                      "end": 104,
                      "start": 99,
                    },
                    "value": "edges",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "loc": Object {
                      "end": 234,
                      "start": 104,
                    },
                    "selections": Array [
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "loc": Object {
                          "end": 208,
                          "start": 134,
                        },
                        "name": Object {
                          "kind": "Name",
                          "loc": Object {
                            "end": 138,
                            "start": 134,
                          },
                          "value": "node",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "loc": Object {
                            "end": 208,
                            "start": 138,
                          },
                          "selections": Array [
                            Object {
                              "alias": null,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "loc": Object {
                                "end": 178,
                                "start": 172,
                              },
                              "name": Object {
                                "kind": "Name",
                                "loc": Object {
                                  "end": 178,
                                  "start": 172,
                                },
                                "value": "field1",
                              },
                              "selectionSet": null,
                            },
                            Object {
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                            },
                          ],
                        },
                      },
                    ],
                  },
                },
              ],
            },
          },
          "type": "Model1Connection",
        },
        Object {
          "args": Array [],
          "fields": Array [],
          "isConnection": false,
          "isFragment": false,
          "isNode": false,
          "name": "id",
          "node": Object {
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "id",
            },
          },
          "type": "ID!",
        },
      ],
      "isConnection": false,
      "isNode": true,
      "name": "viewer",
      "node": Object {
        "alias": null,
        "arguments": Array [],
        "directives": Array [],
        "kind": "Field",
        "loc": Object {
          "end": 274,
          "start": 39,
        },
        "name": Object {
          "kind": "Name",
          "loc": Object {
            "end": 45,
            "start": 39,
          },
          "value": "viewer",
        },
        "selectionSet": Object {
          "kind": "SelectionSet",
          "loc": Object {
            "end": 274,
            "start": 45,
          },
          "selections": Array [
            Object {
              "alias": null,
              "arguments": Array [],
              "directives": Array [],
              "kind": "Field",
              "loc": Object {
                "end": 256,
                "start": 67,
              },
              "name": Object {
                "kind": "Name",
                "loc": Object {
                  "end": 73,
                  "start": 67,
                },
                "value": "model1",
              },
              "selectionSet": Object {
                "kind": "SelectionSet",
                "loc": Object {
                  "end": 256,
                  "start": 73,
                },
                "selections": Array [
                  Object {
                    "alias": null,
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "loc": Object {
                      "end": 234,
                      "start": 99,
                    },
                    "name": Object {
                      "kind": "Name",
                      "loc": Object {
                        "end": 104,
                        "start": 99,
                      },
                      "value": "edges",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "loc": Object {
                        "end": 234,
                        "start": 104,
                      },
                      "selections": Array [
                        Object {
                          "alias": null,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "loc": Object {
                            "end": 208,
                            "start": 134,
                          },
                          "name": Object {
                            "kind": "Name",
                            "loc": Object {
                              "end": 138,
                              "start": 134,
                            },
                            "value": "node",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "loc": Object {
                              "end": 208,
                              "start": 138,
                            },
                            "selections": Array [
                              Object {
                                "alias": null,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "loc": Object {
                                  "end": 178,
                                  "start": 172,
                                },
                                "name": Object {
                                  "kind": "Name",
                                  "loc": Object {
                                    "end": 178,
                                    "start": 172,
                                  },
                                  "value": "field1",
                                },
                                "selectionSet": null,
                              },
                              Object {
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "id",
                                },
                              },
                            ],
                          },
                        },
                      ],
                    },
                  },
                ],
              },
            },
            Object {
              "kind": "Field",
              "name": Object {
                "kind": "Name",
                "value": "id",
              },
            },
          ],
        },
      },
      "type": "Viewer",
    },
  ],
  "text": "query Q1 {
  viewer {
    model1 {
      edges {
        node {
          field1
          id
        }
      }
    }
    id
  }
}
",
  "type": "connection",
}
`;
